samples/bluetooth/st_ble_sensor/src/main.c

```diff
@@ -0,0 +1,185 @@
+ * Copyright (c) 2019 Marcio Montenegro <mtuxpe@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <zephyr.h>
+#include <drivers/gpio.h>
+#include <logging/log.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include "button_svc.h"
+#include "led_svc.h"
+
+LOG_MODULE_REGISTER(main);
+
+extern u16_t but_val;
+extern struct device *led_dev;
+extern bool led_state;
+
+/* Prototype */
+static ssize_t recv(struct bt_conn *conn,
+		    const struct bt_gatt_attr *attr, const void *buf,
+		    u16_t len, u16_t offset, u8_t flags);
+
+/* ST Custom Service  */
+static struct bt_uuid_128 st_service_uuid = BT_UUID_INIT_128(
+	0x8f, 0xe5, 0xb3, 0xd5, 0x2e, 0x7f, 0x4a, 0x98,
+	0x2a, 0x48, 0x7a, 0xcc, 0x40, 0xfe, 0x00, 0x00);
+
+/* ST LED service */
+static struct bt_uuid_128 led_char_uuid = BT_UUID_INIT_128(
+	0x19, 0xed, 0x82, 0xae, 0xed, 0x21, 0x4c, 0x9d,
+	0x41, 0x45, 0x22, 0x8e, 0x41, 0xfe, 0x00, 0x00);
+
+/* ST Notify button service */
+static struct bt_uuid_128 but_notif_uuid = BT_UUID_INIT_128(
+	0x19, 0xed, 0x82, 0xae, 0xed, 0x21, 0x4c, 0x9d,
+	0x41, 0x45, 0x22, 0x8e, 0x42, 0xfe, 0x00, 0x00);
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+#define ADV_LEN 12
+
+/* Advertising data */
+static u8_t manuf_data[ADV_LEN] = {
+	0x01 /*SKD version */,
+	0x83 /* STM32WB - P2P Server 1 */,
+	0x00 /* GROUP A Feature  */,
+	0x00 /* GROUP A Feature */,
+	0x00 /* GROUP B Feature */,
+	0x00 /* GROUP B Feature */,
+	0x00, /* BLE MAC start -MSB */
+	0x00,
+	0x00,
+	0x00,
+	0x00,
+	0x00, /* BLE MAC stop */
+};
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+	BT_DATA(BT_DATA_MANUFACTURER_DATA, manuf_data, ADV_LEN)
+};
+
+/* BLE connection */
+struct bt_conn *p_conn;
+/* Notification state */
+volatile u8_t notify_enable;
+
+static void mpu_ccc_cfg_changed(const struct bt_gatt_attr *attr, u16_t value)
+{
+	ARG_UNUSED(attr);
+	notify_enable = (value == BT_GATT_CCC_NOTIFY);
+	LOG_INF("Notification %s", notify_enable ? "enabled" : "disabled");
+}
+
+/* The STM32WB55xx board is acting as GATT server.
+ * The ST BLE Android app is the BLE GATT client.
+ */
+
+/* ST BLE Sensor GATT services and characteristic */
+
+BT_GATT_SERVICE_DEFINE(stsensor_svc,
+BT_GATT_PRIMARY_SERVICE(&st_service_uuid),
+BT_GATT_CHARACTERISTIC(&led_char_uuid.uuid,
+	BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE_WITHOUT_RESP,
+	BT_GATT_PERM_WRITE, NULL, recv, (void *)1),
+BT_GATT_CHARACTERISTIC(&but_notif_uuid.uuid, BT_GATT_CHRC_NOTIFY,
+	BT_GATT_PERM_READ, NULL, NULL, &but_val),
+BT_GATT_CCC(mpu_ccc_cfg_changed,
+	BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
```

Comment:
And here, though in this case you might be able to fit everything on a single line